File: 8kyu-Counting-Sheep.js
// Consider an array/list of sheep where some sheep may be missing from their place. We need a function that counts the number of sheep present in the array (true means present).

// For example,

// [true,  true,  true,  false,
//   true,  true,  true,  true ,
//   true,  false, true,  false,
//   true,  false, false, true ,
//   true,  true,  true,  true ,
//   false, false, true,  true]
// The correct answer would be 17.

// Hint: Don't forget to check for bad values like null/undefined

describe("Tests", () => {
    it("test", () => {
  var array1 = [true,  true,  true,  false,
                true,  true,  true,  true ,
                true,  false, true,  false,
                true,  false, false, true ,
                true,  true,  true,  true ,
                false, false, true,  true ];
                
  Test.assertEquals(countSheeps(array1), 17, "There are 17 sheeps in total")
    });
  });

// my code

function countSheeps(arrayOfSheep) {
    let count = 0;
    for (let i=0;i<arrayOfSheep.length;i++){
      if (arrayOfSheep[i] == true) {
        count ++;
      } 
    }
    return count
  }

File: CodeWars/.cs/8kyu-CountingSheep.cs
// Consider an array/list of sheep where some sheep may be missing from their place. We need a function that counts the number of sheep present in the array (true means present).

// For example,

// [true,  true,  true,  false,
//   true,  true,  true,  true ,
//   true,  false, true,  false,
//   true,  false, false, true ,
//   true,  true,  true,  true ,
//   false, false, true,  true]
// The correct answer would be 17.

// Hint: Don't forget to check for bad values like null/undefined

using System;
using NUnit.Framework;

[TestFixture]
public class CountSheepsTests {

  [Test]
  public void SampleTest() {
    var sheeps = new bool[] { true, false, true };
    
    Assert.AreEqual(2, Kata.CountSheeps(sheeps));
  }
}

// my code

public static class Kata
{
  public static int CountSheeps(bool[] sheeps)
  {
    int count = 0;
    for (int i = 0; i < sheeps.Length;i++) {
      if (sheeps[i] == true) {count++;}
    }
    return count;
  }
}